vault/resource_mongodbatlas_secret_role.go

```diff
@@ -0,0 +1,238 @@
+	}
+	if v, ok := d.GetOk("ip_addresses"); ok {
+		data["ip_addresses"] = v
+	}
+	if v, ok := d.GetOk("cidr_blocks"); ok {
+		data["cidr_blocks"] = v
+	}
+	if v, ok := d.GetOk("project_roles"); ok {
+		data["project_roles"] = v
+	}
+	if v, ok := d.GetOk("ttl"); ok {
+		data["ttl"] = v
+	}
+	if v, ok := d.GetOk("max_ttl"); ok {
+		data["max_ttl"] = v
+	}
+
+	if _, err := client.Logical().Write(path, data); err != nil {
+		return diag.Errorf("error updating MongoDB Atlas role %q, err=%s", name, err)
+	}
+
+	d.SetId(path)
+
+	return mongodbAtlasSecretRoleRead(ctx, d, meta)
+}
+
+func mongodbAtlasSecretRoleRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	path := d.Id()
+	if err := d.Set(consts.FieldPath, path); err != nil {
+		return diag.FromErr(err)
+	}
+	log.Printf("[DEBUG] Reading MongoDB Atlas role at %q", path)
+
+	resp, err := client.Logical().ReadWithContext(ctx, path)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error reading MongoDB Atlas role at %s, err=%w", path, err))
+	}
+	if resp == nil {
+		log.Printf("[WARN] MongoDB Atlas role not found, removing from state")
+		d.SetId("")
+
+		return diag.FromErr(fmt.Errorf("expected role at %s, no role found", path))
+	}
+
+	if v, ok := resp.Data["name"]; ok {
+		if err := d.Set("name", v); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'name': %s", err))
+		}
+	}
+	if v, ok := resp.Data["organization_id"]; ok {
+		if err := d.Set("organization_id", v); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'organization_id': %s", err))
+		}
+	}
+	if v, ok := resp.Data["project_id"]; ok {
+		if err := d.Set("project_id", v); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'project_id': %s", err))
+		}
+	}
+	if v, ok := resp.Data["roles"]; ok {
+		if err := d.Set("roles", v); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'roles': %s", err))
+		}
+	}
+	if v, ok := resp.Data["ip_addresses"]; ok {
+		if err := d.Set("ip_addresses", v); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'ip_addresses': %s", err))
+		}
+	}
+	if v, ok := resp.Data["cidr_blocks"]; ok {
+		if err := d.Set("cidr_blocks", v); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'cidr_blocks': %s", err))
+		}
+	}
+	if v, ok := resp.Data["project_roles"]; ok {
+		if err := d.Set("project_roles", v); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'project_roles': %s", err))
+		}
+	}
+	if v, ok := resp.Data["ttl"]; ok {
+		if err := d.Set("ttl", v); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'ttl': %s", err))
+		}
+	}
+	if v, ok := resp.Data["max_ttl"]; ok {
+		if err := d.Set("max_ttl", v); err != nil {
+			return diag.FromErr(fmt.Errorf("error setting state key 'max_ttl': %s", err))
+		}
+	}
```

Comment:
Similar comment as above. We can make a list of API fields and loop over them as done [here](https://github.com/hashicorp/terraform-provider-vault/blob/main/vault/resource_kubernetes_secret_backend_role.go#L200-L218)